### ISSUE TITLE
Release 12.2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5879]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5879|
|Type|task|
|Created|2012-08-17 22:24:42Z|
|Branch|https://github.com/twisted/twisted/tree/releases/release-12.2.0-5879|

As of r35278, trunk is in a releasable state as far as I know.

The release notes should state that next version of Twisted (i.e. after 12.2) will begin requiring zope.interface 3.6.

It would be nice to have this branched ASAP, since we have various Python 3 boostrapping code that either will require zope.interface 3.6, or will temporarily make our code less pleasant.

<details><summary>Searchable metadata</summary>

```
trac-id__5879 5879
type__task task
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_releases_release_12_2_0_5879 branches/releases/release-12.2.0-5879
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1345242282000000 1345242282000000
changetime__1346510069000000 1346510069000000
version__None None
owner__ 
cc__therve cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_2"></a>|@ashfall removed owner|
|-|-|

Not sure where in the NEWS file the zope.interface 3.6 requirement from the next release should be mentioned...
Other than that, I think this is ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall commented|
|-|-|

Feedback from IRC on the zope.interface 3.6 requirement announcement in the NEWS file: 

> <tos9> Maybe "as part of python 3 support" makes it sound like that's a thing that will be present in the next release too.


Suggestions for better phrasing welcome. (s/support/preparation, maybe?)

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall set owner to @ashfall|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_5"></a>|@ashfall removed owner|
|-|-|

As suggested, added a note about this being the last release to support Python 2.6 on Windows.

Please review!

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_6"></a>|@ashfall commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tmp-release-12.2.0-5879)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set owner to @ashfall|
|-|-|

Thanks for working on the release! Some minor cleanups:

  1. There's a merge conflict with doc/fun/Twisted.Quotes - some new quotes have been added since you branched, that will need to stay in the file for use in next release (I want the unicode/sockets quote for first release with Python 3 support, personally).
  1. adoptStreamConnection's NEWS has a typo ("connectiong" rather than "connection"), which would need to be fixed in main NEWS file and `twisted/topfiles/NEWS`.
  1. I would move Python 2.5 removal to top of removal list, since it's important (main and core NEWS files).
  1. Description (in main and core NEWS files) for 5632 is missing; maybe just add 5632 to the Other list.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_8"></a>|@ashfall removed owner|
|-|-|

Replying to [itamar](#note_7):
> Thanks for working on the release! Some minor cleanups:
> 
>   1. There's a merge conflict with doc/fun/Twisted.Quotes - some new quotes have been added since you branched, that will need to stay in the file for use in next release (I want the unicode/sockets quote for first release with Python 3 support, personally).

I manually put the new quotes we had since the release branch was cut in doc/fun/Twisted.Quotes. Filed [#5927](https://github.com/twisted/twisted/issues/5927) for this.

2, 3, 4 - done

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [35476]) Merge releases/release-12.2.0-5879

Author: ashfall
Reviewer: itamar
Fixes: [#5879](https://github.com/twisted/twisted/issues/5879)

Release Twisted 12.2.0
